#==================================================================================================
#
# Patch to change the behavior of warning promotion, which is presently enabled for all types of
# debug builds. That includes 'debugoptimized', which one might argue isn't a developer build.
# 
# So, we tweak upstream's logic slightly, only enabling maximum strictness for a pure-debug
# build.
#
# This allows us to avoid having to selectively disable one or more specific warning options,
# since they're no longer promoted to error (via '-Werror=xxx').
#
# Ultimately I plan to collaborate with upstream, to see if they'd be willing to accept this.
# (Either verbatim, or via a slightly different approach.)
#
# Author: Christopher Nielsen
# Date:   2023-09-11
#
#==================================================================================================
--- meson.build.orig	2023-09-11 09:04:47.000000000 -0400
+++ meson.build	2023-09-11 09:05:52.000000000 -0400
@@ -40,7 +40,7 @@
 revision = 0
 libversion = '@0@.@1@.@2@'.format(soversion, current, revision)
 
-gitg_debug = get_option('buildtype').contains('debug')
+gitg_debug = get_option('buildtype') == 'debug'
 
 cc = meson.get_compiler('c')
 valac = meson.get_compiler('vala')
